src/serde/reader.rs

```diff
@@ -46,10 +47,8 @@ pub enum DeserializeError {
     /// If decoding a message takes more than the provided size limit, this
     /// error is returned.
     SizeLimit,
-    SyntaxError(String),
-    EndOfStreamError,
-    UnknownFieldError,
-    MissingFieldError,
+    EndOfStream,
```

Comment:
That's true, but outside of serde, you have `ByteOrderError::UnexpectedEndOfStream`.  I thought it would be weird to coerce a ByteOrderError into a `serde::de::value::Error`, but I could be convinced otherwise.